pkg/prometheus/server/statefulset.go

```diff
@@ -247,13 +246,7 @@ func makeStatefulSetSpec(
 
 	var ports []v1.ContainerPort
 	if !cpf.ListenLocal {
```

Comment:
Similar to simon's comment above, let's move this if inside makecontainerports()